src/docker/remote.rs

```diff
@@ -9,35 +9,40 @@ use eyre::Context;
 use super::engine::Engine;
 use super::shared::*;
 use crate::cargo::CargoMetadata;
-use crate::config::{bool_from_envvar, Config};
+use crate::config::bool_from_envvar;
 use crate::errors::Result;
 use crate::extensions::CommandExt;
 use crate::file::{self, PathExt, ToUtf8};
 use crate::rustc::{self, VersionMetaExt};
 use crate::rustup;
-use crate::shell::{self, MessageInfo, Stream};
+use crate::shell::{ColorChoice, MessageInfo, Stream, Verbosity};
 use crate::temp;
 use crate::{Host, Target};
 
 // the mount directory for the data volume.
 pub const MOUNT_PREFIX: &str = "/cross";
 
-struct DeleteVolume<'a>(&'a Engine, &'a VolumeId, MessageInfo);
+// it's unlikely that we ever need to erase a line in the destructors,
+// and it's better than keep global state everywhere, or keeping a ref
+// cell which could have already deleted a line
```

Comment:
The deleters are the only location we don't have a single instance of `MessageInfo`, but I think this is fine since it avoids having to pass a `RefCell` everywhere, worrying about avoiding multiple mutable borrows, etc., and the chance we delete a line when invoking the deleters is near 0.